spec/core/config_mixin_spec.rb

```diff
@@ -0,0 +1,73 @@
+require "logstash/config/mixin"
+require "logstash/filters/base"
+
+describe LogStash::Config::Mixin do
+  context "when validating :bytes" do
+    it "should validate :bytes" do
+      conf_class = Class.new(LogStash::Filters::Base) do
+        include LogStash::Config::Mixin
+        config_name "test"
+        milestone 1
+        config :size_file, :validate => :bytes
+      end.new({"size_file" => "10Kib"})
+      expect(conf_class.size_file).to eq(10240)
+    end
+
+    it "should allow setting valid default" do
+      conf_class = Class.new(LogStash::Filters::Base) do
+        include LogStash::Config::Mixin
+        config_name "test"
+        milestone 1
+        config :size_file, :validate => :bytes, :default => "10Kib"
+      end.new({})
+      expect(conf_class.size_file).to eq(10240)
+    end
+
+    it "should accept bytes with no units specified in suffix" do
+      conf_class = Class.new(LogStash::Filters::Base) do
+        include LogStash::Config::Mixin
+        config_name "test"
+        milestone 1
+        config :size_file, :validate => :bytes, :default => "1024"
+      end.new({})
+      expect(conf_class.size_file).to eq(1024)
+    end
+
+    it "should be case-insensitive" do
+      conf_class = Class.new(LogStash::Filters::Base) do
+        include LogStash::Config::Mixin
+        config_name "test"
+        milestone 1
+        config :size_file_1, :validate => :bytes, :default => "10KIB"
+        config :size_file_2, :validate => :bytes, :default => "10kib"
+      end.new({})
+      expected_bytes = 10240
+      expect(conf_class.size_file_1).to eq(expected_bytes)
+      expect(conf_class.size_file_2).to eq(expected_bytes)
+    end
+
+    it "should be space-insensitive" do
+      conf_class = Class.new(LogStash::Filters::Base) do
```

Comment:
I just learned that you can give name to your `subject`. see https://relishapp.com/rspec/rspec-core/v/3-1/docs/subject/explicit-subject#use-`subject(:name)`-to-define-a-memoized-helper-method

```ruby
describe "named subject" do
  subject(:bank) { Bank.new }

  it "doesnt have any money" do
    expect(bank.balance).to eq(0)
  end
end
```